File: solutions/scala/bob/2/src/main/scala/Bob.scala
object Bob {
  def response(msg: String): String = {
    val trimmed = msg.trim

    if trimmed.isEmpty then
      return "Fine. Be that way!"

    val isShouting = trimmed.exists(_.isLetter) && trimmed == trimmed.toUpperCase
    val isQuestion = trimmed.endsWith("?")

    (isShouting, isQuestion) match {
      case (true, true) => "Calm down, I know what I'm doing!"
      case (true, false) => "Whoa, chill out!"
      case (false, true) => "Sure."
      case _ => "Whatever."
    }
  }
}

File: solutions/csharp/bob/2/Bob.cs
public static class Bob
{
    public static string Response(string statement)
    {
        if( IsSilence(statement) ) {
            return "Fine. Be that way!";
        }
        
        var trimmedStatement = statement.Trim();

        return (IsYelling(trimmedStatement), IsQuestion(trimmedStatement)) switch
        {
            (true, true) => "Calm down, I know what I'm doing!",
            (true, false) => "Whoa, chill out!",
            (false, true) => "Sure.",
            _ => "Whatever."
        };
    }

    private static bool IsSilence(string statement) => string.IsNullOrWhiteSpace(statement);

    private static bool IsQuestion(string statement) => statement.EndsWith('?');

    private static bool IsYelling(string statement) =>
        statement.Any(char.IsLetter) && !statement.Any(char.IsLower);
}
